jaeger_client/tracer.py

```diff
@@ -157,10 +157,12 @@ def start_span(self,
                     tags = tags or {}
                     for k, v in six.iteritems(sampler_tags):
                         tags[k] = v
-            elif self.is_debug_allowed(operation_name):  # have debug id
+            elif parent.debug_id and self.is_debug_allowed(operation_name):
                 flags = SAMPLED_FLAG | DEBUG_FLAG
                 tags = tags or {}
                 tags[self.debug_id_header] = parent.debug_id
+            if parent and parent.baggage:
+                baggage = dict(parent.baggage)  # TODO do we need to clone?
```

Comment:
If the `dict` is immutable, you should not have to clone. Immutability is definitely the approach in other clients, IDK about this one.